src/sentry/flags/models.py

```diff
@@ -0,0 +1,65 @@
+from enum import Enum
+
+from django.db import models
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import Model, region_silo_model, sane_repr
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+
+
+class ActionEnum(Enum):
+    CREATED = 0
+    DELETED = 1
+    UPDATED = 2
+
+
+ACTION_MAP = {
+    "created": ActionEnum.CREATED.value,
+    "deleted": ActionEnum.DELETED.value,
+    "updated": ActionEnum.UPDATED.value,
+}
+
+
+class ModifiedByTypeEnum(Enum):
+    EMAIL = 0
+    ID = 1
+    NAME = 2
+
+
+MODIFIED_BY_TYPE_MAP = {
+    "email": ModifiedByTypeEnum.EMAIL.value,
+    "id": ModifiedByTypeEnum.ID.value,
+    "name": ModifiedByTypeEnum.NAME.value,
+}
+
+
+@region_silo_model
+class FlagAuditLogModel(Model):
```

Comment:
Optional: You can use `DefaultFieldsModel` to get default fields for when this row was created/updated

Reply:
it has a downside though -- its "default" fields are nullable which makes them very annoying to use

Reply:
I've fixed that. `DefaultFieldsModelExisting` is the old base class with nullable fields, which if we want we could slowly migrate old models off of. `DefaultFieldsModel` has non-nullable cols

Reply:
onice!

Reply:
I'll leave it the way it is for now since I want people (read: me) to get an error when they forget to specify a column in their implementation.  Explicit over implicit and all that.